tasks/20.txt

```diff
@@ -1,3 +1,33 @@
 Create a SQL query to find the employee that sold products for the largest amount:
 
-| EmployeeId | Employee Full Name | Amount, $ |
\ No newline at end of file
+| EmployeeId | Employee Full Name | Amount, $ |
+
+//task20
+ db.employees.aggregate([
+        { $limit : 1 },
+        { $lookup:
+           {
+             from: "orders",
+             localField: "EmployeeID",
+             foreignField: "EmployeeID",
+             as: "orders_obj"
+           }        
+       },
+       { $unwind : "$orders_obj" },
+       
+       { $lookup:
+           {
+             from: "order-details",
+             localField: "orders_obj.OrderID",
+             foreignField: "OrderID",
+             as: "order_details_obj"
+           }        
+       },
+       { $unwind : "$order_details_obj" },
+       { $group:
+         {
+           _id: { EmployeeID: "$EmployeeID", FullName: { $concat: ["$FirstName", " ", "$LastName"] } }, 
+           Amount: { $max: { $multiply: [ "$order_details_obj.UnitPrice", "$order_details_obj.Quantity" ] } }
+         }
+     }    
+ ])
```

Comment:
We also have to consider a discount. Also I am not sure why $limit is a first step?

Reply:
Logic is incorrect now because count and limit do not consider a discount.
We can calculate final Amount at the $group stage.  That have to resolve the issue